editor/project_manager.cpp

```diff
@@ -1325,6 +1326,21 @@ void ProjectManager::_erase_project() {
 	erase_ask->popup_centered_minsize();
 }
 
+void ProjectManager::_language_selected(int p_id) {
+
+	String lang = language_btn->get_item_metadata(p_id);
+	EditorSettings::get_singleton()->set("interface/editor/editor_language", lang);
+	language_btn->set_text(lang);
+
+	List<String> args = OS::get_singleton()->get_cmdline_args();
+	String exec = OS::get_singleton()->get_executable_path();
+	OS::ProcessID pid = 0;
+	Error err = OS::get_singleton()->execute(exec, args, false, &pid);
+	ERR_FAIL_COND(err);
+
+	get_tree()->quit();
```

Comment:
That seems a bit brutal if that happens right after language selection without warning. I think it would be better to show a popup mentioning that the engine has to be restarted for the language selection to be applied, with options to Restart or Continue (since opening a project would change the language anyway, personally I'd likely opt for continuing without a restart of the project manager, and just open my project to get settings applied).

Reply:
Good point, a confirmation dialog would be less of an interruption.
<img width="602" alt="confirm restart" src="https://user-images.githubusercontent.com/11854330/32073615-412b8bee-ba5c-11e7-982d-2b8cea6fcf94.PNG">